packages/financial-templates-lib/src/logger/DiscordTicketTransport.ts

```diff
@@ -89,6 +106,39 @@ export class DiscordTicketTransport extends Transport {
 
   // Use bot token for establishing a connection to Discord API.
   async login(): Promise<void> {
+    this.isFlushed = false; // Set the isFlushed to false to prevent the logger from closing while logging in.
     await this.client.login(this.botToken);
+    this.isFlushed = true;
+  }
+
+  async executeLogQueue(): Promise<void> {
+    if (this.isQueueBeingExecuted) return; // If the queue is currently being executed, return.
+    this.isQueueBeingExecuted = true; // Lock the queue to being executed.
```

Comment:
+1 to queue locking to keep the timer coherent. This will never miss messages _as long as_ there are no awaits _after_ the error handling or falling off the end of the loop.